fastar/rules.py

```diff
@@ -250,3 +248,36 @@ def dot_general_backward_rule(outbox, lhs, rhs, dimension_numbers, precision):
   return [lhs_start, rhs_start], [lhs_incount, rhs_incount]
 
 def_standard_op(lax.dot_general_p, dot_general_backward_rule)
+
+def pad_backward_rule(outbox, operand, padding_value, padding_config):
+  out_start, out_shape = outbox
+  lo, _, interior = unzip3(padding_config)
```

Comment:
One slight optimization we could do here is to catch when the outbox only contains padded values (i.e. it has no dependence on the input data) and return `None` for the operand instart and count. That case would also have a simpler update rule.

Reply:
Done!